Nef_3/include/CGAL/Nef_3/bounded_side_3.h

```diff
@@ -70,21 +70,27 @@ Bounded_side bounded_side_3(IteratorForward first,
 
 
 
+  auto apv = approx(plane.orthogonal_vector());
+
+  int dir = 0;
+  auto max = CGAL::abs(apv.x());
+  if(CGAL::abs(apv.y()) > max){
+    dir = 1;
+    max = CGAL::abs(apv.y());
+  }
+  if(CGAL::abs(apv.z()) > max){
```

Comment:
@maxGimeno I get the exception: terminate called after throwing an instance of 'CGAL::Uncertain_conversion_exception' what(): Undecidable conversion of CGAL::Uncertain<T>

The stack trace shows the exception coming from line 81 of bounded_side_3.h

Reply:
@afabri Would this approach have the performance benefit:

```c++
  CGAL_assertion(!plane.is_degenerate());
  Point_2 (*t)(const Point_3&);
  Vector_3 pv(plane.orthogonal_vector());
#if CGAL_USE_TRACE
  Vector_3 pxy(0,0,1), pyz(1,0,0), pzx(0,1,0);
#endif
  CGAL_NEF_TRACEN("pv*pxz: "<<pv*pzx);
  CGAL_NEF_TRACEN("pv*pyz: "<<pv*pyz);
  CGAL_NEF_TRACEN("pv*pxy: "<<pv*pxy);
  if( !CGAL_NTS is_zero(pv.y()) )
    /* the plane is not perpendicular to the ZX plane */
    t = &point_3_get_z_x_point_2< Point_2, Point_3>;
  else if( !CGAL_NTS is_zero(pv.x()) )
    /* the plane is not perpendicular to the YZ plane */
    t = &point_3_get_y_z_point_2< Point_2, Point_3>;
  else {
    CGAL_assertion( !CGAL_NTS is_zero(pv.z()) );
    /* the plane is not perpendicular to the XY plane */
    t = &point_3_get_x_y_point_2< Point_2, Point_3>;
  }
  ```

Reply:
Hmm, although `is_zero` uses exact for the calculation what we actually want is `is_non_zero` which I think could use interval math?

Reply:
I think using simply `pv.y() != 0`  instead of `!CGAL_NTS is_zero(pv.y())` achieves this for intervals.

Reply:
My mistake was to compare intervals.   In case they overlap, that is for exampe   `a.inf() < b.inf() < a.sup()` the exception is thrown.

Reply:
> I think using simply `pv.y() != 0` instead of `!CGAL_NTS is_zero(pv.y())` achieves this for intervals.

`is_zero()`   is a better choice. For example with a number type like `Gmpq`, it is pointless to constuct the quotient `0/1` and to do the crosswise multiplication with the denominators, but it is enough to test if the numerator is zero.

Reply:
Oh my solution doesn't take `approx(plane.orthogonal_vector())` Well do you want me to separate out the homogeneous/cartesian changes I did?